assets/javascripts/kitten/components/form/text-input-with-button/index.js

```diff
@@ -10,115 +11,108 @@ import deprecated from 'prop-types-extra/lib/deprecated'
 
 const StyledTextInputWithButton = styled.div`
   display: flex;
-`
 
-const StyledTextInput = styled(TextInput)`
-  border-right: 0;
-`
+  .k-Form-TextInputWithButton--input {
+    border-right: 0;
+  }
 
-const StyledButton = styled.button`
-  ${TYPOGRAPHY.fontStyles.regular};
-  font-size: ${stepToRem(-1)};
-  line-height: 1.3;
-  flex: 1 0 auto;
-  appearance: none;
-  cursor: pointer;
-  padding: 0 ${pxToRem(30)};
-  border-radius: 0;
+  .k-Form-TextInputWithButton--button {
+    ${TYPOGRAPHY.fontStyles.regular};
+    font-size: ${stepToRem(-1)};
+    line-height: 1.3;
+    flex: 1 0 auto;
+    appearance: none;
+    cursor: pointer;
+    padding: 0 ${pxToRem(30)};
+    border-radius: 0;
 
-  ${({ modifier }) => modifierStyles(modifier)};
+    ${({ modifier }) => modifierStyles(modifier)};
+  }
 
-  ${({ valid }) =>
-    valid &&
-    css`
-      background-color: ${COLORS.valid};
-      border-color: ${COLORS.valid};
-    `}
+  &.k-Form-TextInputWithButton--button__valid {
+    background-color: ${COLORS.valid};
+    border-color: ${COLORS.valid};
+  }
+      
+  &.k-Form-TextInputWithButton--button__error {
+    background-color: ${COLORS.error};
+    border-color: ${COLORS.error};
 
-  ${({ error }) =>
-    error &&
-    css`
+    input:invalid:not(:focus) + & {
       background-color: ${COLORS.error};
       border-color: ${COLORS.error};
-    `}
-
-  input:invalid:not(:focus) + & {
-    background-color: ${COLORS.error};
-    border-color: ${COLORS.error};
+    }
+  }
+  
+  &.k-Form-TextInputWithButton--button__error {
+    cursor: not-allowed;
   }
-
-  ${({ disabled }) =>
-    disabled &&
-    css`
-      cursor: not-allowed;
-    `}
 `
 
-export class TextInputWithButton extends PureComponent {
-  static propTypes = {
-    valid: PropTypes.bool,
-    error: PropTypes.bool,
-    disabled: PropTypes.bool,
-    tiny: PropTypes.bool,
-    huge: PropTypes.bool,
-    giant: PropTypes.bool,
-    modifier: PropTypes.string,
-    inputValue: PropTypes.string,
-    // DEPRECATED: do not use prop `value`. Use `buttonValue` instead.
-    value: deprecated(PropTypes.node, 'Use `buttonValue` instead.'),
-    buttonValue: PropTypes.node,
-  }
+export const TextInputWithButton = ({
+  valid,
+  error,
+  disabled,
+  size,
+  value,
+  buttonValue,
+  modifier,
+  buttonProps,
+  inputValue,
+  className,
+  ...others
+}) => {
 
-  static defaultProps = {
-    valid: false,
-    error: false,
-    disabled: false,
-    tiny: false,
-    huge: false,
-    giant: false,
-    buttonValue: 'Button',
-    modifier: 'beryllium',
-  }
+  return (
+    <StyledTextInputWithButton modifier={modifier}>
```

Comment:
Je mettrais une class sur le wrapper

```suggestion
    <StyledTextInputWithButton className="k-Form-TextInputWithButton" modifier={modifier}>
```